### ISSUE TITLE
Can't do basic POS tagging because PyText's WordTaggingTask expects int:int:string labels

### ISSUE THREAD

**Issue by clefourrier** [facebookresearch/pytext#219]
I am trying to do a simple POS tagging task, for which I am using CONLL data files (without blank or commented lines), and a JSON configuration file calling WordTaggingTask per the coding documentation. (See end of issue for complete configuration file). It does not detect the labels of the dataset properly.

## Steps to reproduce
- Download CONLL data files (such as [here](https://github.com/UniversalDependencies/UD_French-GSD/blob/master/fr_gsd-ud-train.conllu)).
- Remove all commented lines and blank lines from the file
- Create a configuration file (using the content provided below)
- Launch training with `pytext train config_file.json`.

## Observed Results

When launching the training, everything runs without error, but only 2 labels are detected (I actually only found "NoLabel" when following the execution) when around 20 are expected. 

```
Building vocab for feature word_feat from train data only.
word_feat field's vocabulary size is 39587
Building vocab for label word_label
word_label field's vocabulary size is 2
```

Of course, the model cannot train on 2 labels, and I get the following at each iteration.
```
Stage.TRAIN
loss: 0.008701
	Per label scores                        	Precision 	Recall    	F1        	Support   

	PAD_LABEL                               	0.00      	0.00      	0.00      	0         

	Overall micro scores                    	0.00      	0.00      	0.00      	0         
	Overall macro scores                    	0.00      	0.00      	0.00      

Rank 0 worker: Running epoch for Stage.EVAL
```

## Where could it come from?

I followed the code execution to this point:
- DataHandler.preprocess(...) calls JointDataHandler.preprocess_row(...), which calls DataUtils.align_slot_labels(...) cf below
```
        if DatasetFieldName.WORD_LABEL_FIELD in self.labels:
            # TODO move it into word label field
            res[DatasetFieldName.WORD_LABEL_FIELD] = data_utils.align_slot_labels(
                features.token_ranges,
                row_data.get(DFColumn.WORD_LABEL),
                self.labels[DatasetFieldName.WORD_LABEL_FIELD].use_bio_labels,
            )
            res[DatasetFieldName.RAW_WORD_LABEL] = row_data.get(DFColumn.WORD_LABEL)
```
When looking, row_data.get(DFColumn.WORD_LABEL) contains the raw labels. 

- data_utils.align_slot_labels(...) then calls data_utils.parse_slot_string(...) (cf below or l.53 of data_utils.py) which **systematically tries to split the labels and assumes that they are of the form int:int:string**. (Which is not the case for my labels, simple strings that they are).
```
    slots = slots_field.split(",")
    slot_list = []
    for slot in slots:
        slot_toks = slot.split(":", 2)
        if len(slot_toks) == 3:
            curr_slot = Slot(slot_toks[2], int(slot_toks[0]), int(slot_toks[1]))
            slot_list.append(curr_slot)
    return slot_list
```
The snippet above returns an empty slot list, which then is converted to "NoLabel".

## Expected Results

I expected pytext to properly detect simple labels and not assume they are of the shape int:int:string. If this is the expected behaviour, could you explain why?

## Relevant Code

Config_file.json
```json
{
    "task": {
        "WordTaggingTask": {
            "model": {
                "representation": {
                    "BiLSTMSlotAttention": {
                        "dropout": 0,
                        "lstm": {
                            "dropout": 0,
                            "lstm_dim": 100,
                            "num_layers": 1,
                            "bidirectional": true
                        }
                    }
                },
                "output_layer": {
                    "WordTaggingOutputLayer": {}
                }
            },
            "optimizer": {
                "type": "adam",
                "lr": "0.001"
            },
            "trainer": {
                "epochs": 2
            },
            "data_handler": {
                "columns_to_read": [
                    "other_info",
                    "text",
                    "other_info",
                    "word_label",
                    "other_info0",
                    "other_info1",
                    "other_info2",
                    "other_info3",
                    "other_info4",
                    "other_info5"
                ],
                "train_path": "../datasets/fr_gsd-ud-train.conllu",
                "eval_path": "../datasets/fr_gsd-ud-dev.conllu",
                "test_path": "../datasets/fr_gsd-ud-test.conllu",
                "train_batch_size": 128,
                "eval_batch_size": 128,
                "test_batch_size": 128
            },
            "labels": {
                "use_bio_labels": false
            }
        }
    }
}

```

**Comment by clefourrier** [comment]
Please tell me if there is any further information you might need to reproduce the bug.

**Comment by Titousensei** [comment]
The new PyText version (released as 0.2.0) should give you the capabilities to do that. Look at the documentation for "Custom Data Format" and "Creating A New Model", that looks like what you're asking.
